internals/types/index.d.ts

```diff
@@ -1,6 +1,12 @@
 import type { BuilderFunctions } from '@vercel/build-utils';
 import type { Readable, Writable } from 'stream';
 import type { Route } from '@vercel/routing-utils';
+import { PROJECT_ENV_TARGET, PROJECT_ENV_TYPE } from '@vercel-internals/constants';
+
+export type ValueOf<T> = T[keyof T];
+
+export type PROJECT_ENV_TARGET_VALUES = ValueOf<PROJECT_ENV_TARGET>;
+export type PROJECT_ENV_TYPE_VALUES = ValueOf<PROJECT_ENV_TYPE>;
```

Comment:
Lets use PascalCase for these two types since that is the convention

Reply:
Not for constant values is it? I've always thought constants were upper-underscore cased

Reply:
Its not a constant. That would be `export const`. I'm talking about the `export type`

Reply:
Ahh okay I think i get it. My brain still sees this as _constant types_ since they are based off of the constants, but I also see your point that they are _types_ and not _constants_. Changing!